File: atcoder/abc052B_increment_decrement.nim
# uninhm
# https://atcoder.jp/contests/abc052/tasks/abc052_b
# simulation

discard stdin.readLine
let s = stdin.readLine

var ans = 0
var x = 0
for i in s:
  case i:
    of 'I': inc x
    of 'D': dec x
    else: echo "WTF"
  ans = max(ans, x)

echo ans

File: atcoder/abc052B_increment_decrement.py
# uninhm
# https://atcoder.jp/contests/abc052/tasks/abc052_b
# implementation

input()
s = input()

ans = 0
x = 0
for i in s:
    if i == 'I':
        x += 1
    elif i == 'D':
        x -= 1
    ans = max(ans, x)

print(ans)
